### PR TITLE
docs: imperative vs declarative terminology for Prisma Migrate

### PR DESCRIPTION
**Pull request by ruheni**
Highlight how Prisma Migrate is a hybrid migration tool

Thread: https://prisma-company.slack.com/archives/CGMCB82N8/p1693503016448309

### CODE REVIEW

**Comment by janpio** [comment]
Although correct, it feels wrong to start the documentation with this statement. I think we should start with the features first (as it did before), and then maybe add a slightly academic paragraph on hybrid/imperative/declarative and what that means.

**Comment by ruheni** [comment]
I've re-ordered the content. Let me know if it reads better.

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 1]
```diff
@@ -13,18 +13,18 @@ toc: false
...
 Prisma Migrate enables you to:
 
 - Keep your database schema in sync with your [Prisma schema](/concepts/components/prisma-schema) as it evolves _and_
 - Maintain existing data in your database
 
 Prisma Migrate generates [a history of `.sql` migration files](/concepts/components/prisma-migrate/migration-histories), and plays a role in both [development and deployment](/concepts/components/prisma-migrate/migrate-development-production).
 
+Prisma Migrate is both an **imperative and declarative database schema migration tool** because it:
+
+- Provides declarative data modeling, from which it generates SQL migrations
```
Comment (janpio) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]: This should mention and link to Prisma schema gain

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 2]
```diff
@@ -13,18 +13,18 @@ toc: false
...
+- Allows you to modify the generated SQL migrations before applying them to your database
```
Comment (janpio) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]: Why is the fact that you can modify them relevant here?

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 3]
```diff
@@ -13,18 +13,18 @@ toc: false
...
-- Allows you to modify the generated SQL migrations before applying them to your database
-
 Prisma Migrate enables you to:
 
 - Keep your database schema in sync with your [Prisma schema](/concepts/components/prisma-schema) as it evolves _and_
 - Maintain existing data in your database
 
 Prisma Migrate generates [a history of `.sql` migration files](/concepts/components/prisma-migrate/migration-histories), and plays a role in both [development and deployment](/concepts/components/prisma-migrate/migrate-development-production).
 
+Prisma Migrate is both an **imperative and declarative database schema migration tool** because it:
```
Comment (janpio) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]: Which one of the bullets refers to which here?

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 4]
```diff
@@ -13,13 +13,18 @@ toc: false
...
 
 - Keep your database schema in sync with your [Prisma schema](/concepts/components/prisma-schema) as it evolves _and_
 - Maintain existing data in your database
 
 Prisma Migrate generates [a history of `.sql` migration files](/concepts/components/prisma-migrate/migration-histories), and plays a role in both [development and production](/concepts/components/prisma-migrate/migrate-development-production).
 
+Prisma Migrate is a **hybrid** — imperative and declarative — database schema migration tool:
+
+- Declarative: generates SQL migrations based on your [Prisma schema](/concepts/components/prisma-schema)
+- Imperative: provides flexibility by enabling you to modify what and how they're executed and perform data migrations
```
Comment (jharrell) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]: not sure what "they" we're referring to here. migrations?

Comment (ruheni) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]:
```suggestion
- Imperative: provides flexibility by enabling you to modify what and how migrations are executed and perform data migrations
```

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 5]
```diff
@@ -13,13 +13,18 @@ toc: false
...
 
-Prisma Migrate is an **declarative database schema migration tool** that enables you to:
+Prisma Migrate enables you to:
 
 - Keep your database schema in sync with your [Prisma schema](/concepts/components/prisma-schema) as it evolves _and_
 - Maintain existing data in your database
 
 Prisma Migrate generates [a history of `.sql` migration files](/concepts/components/prisma-migrate/migration-histories), and plays a role in both [development and production](/concepts/components/prisma-migrate/migrate-development-production).
 
+Prisma Migrate is a **hybrid** — imperative and declarative — database schema migration tool:
```
Comment (nikolasburk) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]:
```suggestion
Prisma Migrate can be considered a _hybrid_ database schema migration tool, meaning it has both of _declarative_ and  _imperative_ elements:
```

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 6]
```diff
@@ -13,13 +13,18 @@ toc: false
...
+Prisma Migrate enables you to:
 
 - Keep your database schema in sync with your [Prisma schema](/concepts/components/prisma-schema) as it evolves _and_
 - Maintain existing data in your database
 
 Prisma Migrate generates [a history of `.sql` migration files](/concepts/components/prisma-migrate/migration-histories), and plays a role in both [development and production](/concepts/components/prisma-migrate/migrate-development-production).
 
+Prisma Migrate can be considered a _hybrid_ database schema migration tool, meaning it has both of _declarative_ and  _imperative_ elements:
+
+- Declarative: generates SQL migrations based on your [Prisma schema](/concepts/components/prisma-schema)
```
Comment (nikolasburk) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]:
```suggestion
- Declarative: The data model is described in a declarative way in the [Prisma schema](/concepts/components/prisma-schema). Prisma Migrate generates SQL migration files from that data model.
```

File: content/200-concepts/100-components/03-prisma-migrate/index.mdx [hunk 7]
```diff
@@ -13,13 +13,18 @@ toc: false
...
+- Imperative: provides flexibility by enabling you to modify what and how migrations are executed and perform data migrations
```
Comment (nikolasburk) [inline content/200-concepts/100-components/03-prisma-migrate/index.mdx]:
```suggestion
- Imperative: All generated SQL migration files are fully customizable. Prisma Migrate hence provides the flexibility of an imperative migration tool by enabling you to modify what and how migrations are executed (and allows you to run custom SQL to e.g. make use of native database feature, perform data migrations, ...).
```

**Pull request merged**